File: Web/WeddingPlanner/02-Domain/2.01-Entity/PayObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WeddingPlanner.Domain.Entity
{
    public class PayObject
    {
        public decimal Price { get; set; }
        public decimal Incoming { get; set; }
        public bool SinglePayment { get; set; }
        public int Parcels { get; set; }
        public int ParcelsPaid { get; set; }
        
        public bool Paid {
            get { return RemainingPrice <= 0; }
        }
        public decimal ParcelsPrice {
            get { return Price*((Price - Incoming)/Parcels); }
        }
        public decimal RemainingPrice {
            get { return Price-Incoming-(ParcelsPaid*ParcelsPrice); }
        }
    }
}

File: IOS/WeddingPlanner/PayObject.swift
//
//  PayObject.swift
//  WeddingPlanner
//
//  Created by Felipe Assunção on 4/19/16.
//  Copyright © 2016 Felipe Assunção. All rights reserved.
//

import UIKit
import SwiftMongoDB

public struct PayObject: MongoObject {

    var   price         : NSDecimal?
        , incoming      : NSDecimal?
        , parcels       : Int?
        , parcelsPaid   : Int?
        , parcelsPrice  : NSDecimal?
        , Paid          : Bool?
}
